core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -278,4 +278,39 @@ package object config {
         "spark.io.compression.codec.")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val SHUFFLE_ACCURATE_BLOCK_THRESHOLD =
+    ConfigBuilder("spark.shuffle.accurateBlkThreshold")
+      .doc("When we compress the size of shuffle blocks in HighlyCompressedMapStatus, we will" +
+        "record the size accurately if it's above the threshold specified by this config. This " +
+        "helps to prevent OOM by avoiding underestimating shuffle block size when fetch shuffle " +
+        "blocks.")
+      .longConf
+      .createWithDefault(100 * 1024 * 1024)
+
+  private[spark] val MEMORY_OFF_HEAP_ENABLED =
+    ConfigBuilder("spark.memory.offHeap.enabled")
+      .doc("If true, Spark will attempt to use off-heap memory for certain operations(e.g. sort, " +
+        "aggregate, etc. However, the buffer used for fetching shuffle blocks is always " +
+        "off-heap). If off-heap memory use is enabled, then spark.memory.offHeap.size must be " +
+        "positive.")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val MEMORY_OFF_HEAP_SIZE =
+    ConfigBuilder("spark.memory.offHeap.size")
+      .doc("The absolute amount of memory in bytes which can be used for off-heap allocation." +
+        " This setting has no impact on heap memory usage, so if your executors' total memory" +
+        " consumption must fit within some hard limit then be sure to shrink your JVM heap size" +
+        " accordingly. This must be set to a positive value when " +
+        "spark.memory.offHeap.enabled=true. Note that Blocks will be shuffled to off-heap.")
```

Comment:
`Note that Blocks will be shuffled to off-heap.`  this is not needed as we already mentioned it in `MEMORY_OFF_HEAP_ENABLED`